src/components/Result.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { render, screen } from '@testing-library/react';
+import Result from './Result';
+import { Context } from '../Context';
+import { translationsProvider } from '../util';
+import { QuestionTypes, ResultType } from '../types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const customRender = (ui: any, { providerProps, ...renderOptions }: any) => {
+  return render(<Context.Provider {...providerProps}>{ui}</Context.Provider>, {
+    wrapper: translationsProvider,
+    ...renderOptions,
+  });
+};
+
+describe('Result component', () => {
+  it('renders a reference link', () => {
+    const results: ResultType[] = [
+      {
+        id: '',
+        question: {
+          id: '',
+          points: 0,
+          referenceLink: 'https://www.example.com',
+          type: QuestionTypes.SingleChoice,
+          text: '',
+          answers: [],
+          remainingAttempts: 0,
+          correctlyAnswered: true,
+        },
+      },
+    ];
+
+    const providerProps = {
+      value: {
+        results,
+        numberOfQuestions: 2,
```

Comment:
It does not seem to make a difference, but there is only one question in the dataset. To avoid confusion I'd say we set it to 1 (same below).